src/ORM/Translatable/TranslatableSubscriber.php

```diff
@@ -314,6 +314,10 @@ private function setLocales(LifecycleEventArgs $eventArgs)
         $entity        = $eventArgs->getEntity();
         $classMetadata = $em->getClassMetadata(get_class($entity));
 
+        if (!$this->isTranslatable($entity)) {
+            return;
+        }
+
         if (!$this->getClassAnalyzer()->hasMethod($classMetadata->reflClass, 'setCurrentLocale')) {
```

Comment:
I'd think you can remove this `if`... If you check if the `$entity` is Translatable, the method's availability should be implied from that...